src/Migrator.sol

```diff
@@ -83,44 +83,101 @@ contract Migrator is IMigrator, Initializable, Ownable {
+        Internal functions
+    */
+
+    /**
+     * @notice Calculate the min amount of TOKEN and WETH for a given SLP amount
+     * @param _slpAmount The amount of SLP
+     * @return Return values for min amount out of TOKEN and WETH with unwrap slippage
+     */
+    function _calculateSlpAmounts(uint256 _slpAmount) internal view returns (uint256, uint256) {
+        uint256 tokenPriceEth = _tokenPrice();
+        uint256 wethPriceUsd = _wethPrice();
         uint256 slpSupply = slp.totalSupply();
         (uint256 wethReserves, uint256 tokenReserves, ) = slp.getReserves();
 
-        // Calculate the amount of tokens and WETH the user will receive
-        uint256 amountToken = (_slpAmount * tokenReserves) / slpSupply;
-        uint256 amountWeth = (_slpAmount * wethReserves) / slpSupply;
+        // Convert reserves into current USD price
+        uint256 tokenReservesUsd = ((tokenReserves * tokenPriceEth) * wethPriceUsd) / 1 ether;
+        uint256 wethReservesUsd = wethReserves * wethPriceUsd;
+
+        // Get amounts in USD given the amount of SLP with slippage
+        uint256 amountTokenUsd = (((_slpAmount * tokenReservesUsd) / slpSupply) * (BPS - unrwapSlippage)) / BPS;
+        uint256 amountWethUsd = (((_slpAmount * wethReservesUsd) / slpSupply) * (BPS - unrwapSlippage)) / BPS;
 
-        // Calculate the minimum amounts with unrwapSlippage tolerance
-        uint256 amountTokenMin = (amountToken * (BPS - unrwapSlippage)) / BPS;
-        uint256 amountWethMin = (amountWeth * (BPS - unrwapSlippage)) / BPS;
+        // Return tokens denominated in ETH
+        uint256 amountTokenMin = (amountTokenUsd * 1 ether) / (tokenPriceEth * wethPriceUsd);
+        uint256 amountWethMin = (amountWethUsd) / wethPriceUsd;
 
         return (amountTokenMin, amountWethMin);
     }
 
-    /// @inheritdoc IMigrator
-    function calculateWethWeight(uint256 _tokenAmount) public view returns (uint256) {
-        (, int256 tokenEthPrice, , , ) = priceFeed.latestRoundData();
-
+    /**
+     * @notice Get the amount WETH required to create balanced pool deposit
+     * @param _tokenAmount Amount of TOKEN to deposit
+     * @return uint256 Amount of WETH required to create 80/20 balanced deposit
+     */
+    function _calculateWethWeight(uint256 _tokenAmount) internal view returns (uint256) {
         uint256[] memory normalizedWeights = IManagedPool(address(balancerPool)).getNormalizedWeights();
 
-        uint256 amount = (((_tokenAmount * uint256(tokenEthPrice)) / 1 ether) * normalizedWeights[0]) /
-            normalizedWeights[1];
+        uint256 amount = (((_tokenAmount * _tokenPrice()) / 1 ether) * normalizedWeights[0]) / normalizedWeights[1];
 
         return (amount);
     }
 
-    /// @inheritdoc IMigrator
-    function unwrapSlp() public {
-        uint256 deadline = block.timestamp + 300;
+    /**
+     * @notice Unrwap SLP into TOKEN and WETH
+     */
+    function _unwrapSlp() internal {
+        uint256 deadline = block.timestamp;
```

Comment:
nit: You don't need to make a variable for this, you can just pass in `block.timestamp` directly into the function. You can also use named parameters if you want to specify to readers what `block.timestamp` means.

For example:

```
someFunction({deadline: block.timestamp});
```